src/components/Table/GaiaChainTable.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import { createStyles, Theme, withStyles, WithStyles, makeStyles } from '@material-ui/core/styles';
+import TableCell from '@material-ui/core/TableCell';
+import Paper from '@material-ui/core/Paper';
+import { AutoSizer, Column, Table, TableCellRenderer, TableHeaderProps } from 'react-virtualized';
+
+declare module '@material-ui/core/styles/withStyles' {
+  // Augment the BaseCSSProperties so that we can control jss-rtl
+  interface BaseCSSProperties {
+    /*
+     * Used to control if the rule-set should be affected by rtl transformation
+     */
+    flip?: boolean;
+  }
+}
+
+const styles = (theme: Theme) =>
+  createStyles({
+    head: {
+      backgroundColor: '#F7F8F9',
+      color: theme.palette.common.black,
+      width: '100%',
+    },
+    flexContainer: {
+      display: 'flex',
+      alignItems: 'center',
+      boxSizing: 'border-box',
+    },
+    table: {
+      // temporary right-to-left patch, waiting for
+      // https://github.com/bvaughn/react-virtualized/issues/454
+      '& .ReactVirtualized__Table__headerRow': {
+        flip: false,
+        paddingRight: theme.direction === 'rtl' ? '0 !important' : undefined,
+      },
+    },
+    tableRow: {
+      cursor: 'pointer',
+    },
+    tableRowHover: {
+      '&:hover': {
+        backgroundColor: theme.palette.grey[200],
+      },
+    },
+    tableCell: {
+      flex: 1,
+    },
+    noClick: {
+      cursor: 'initial',
+    },
+  });
+
+interface ColumnData {
+  dataKey: string;
+  label: string;
+  numeric?: boolean;
+  width: number;
+}
+
+interface Row {
+  index: number;
+}
+
+interface MuiVirtualizedTableProps extends WithStyles<typeof styles> {
+  columns: ColumnData[];
+  headerHeight?: number;
+  onRowClick?: () => void;
+  rowCount: number;
+  rowGetter: (row: Row) => Data;
+  rowHeight?: number;
+}
+
+class MuiVirtualizedTable extends React.PureComponent<MuiVirtualizedTableProps> {
```

Comment:
not sure we want to have this as a PureComponent bc we might at some point want a data update tied to a particular lifecycle step on this component (ie: this table will refresh when we switch tabs, rather than having to refresh the page?)

Reply:
<img width="1250" alt="Screenshot 2021-05-20 at 18 36 22" src="https://user-images.githubusercontent.com/16255463/119017561-6e81c580-b99b-11eb-809a-bba887d61a1d.png">